src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php

```diff
@@ -141,7 +179,7 @@ private function createAllComponentsReturnType(bool $urlIsLowercase): Type
 		return $this->allComponentsTogetherType;
 	}
 
-	private function createComponentsArray(bool $urlIsLowercase): Type
+	private function createComponentsArray(bool $urlIsLowercase, bool $urlIsUppercase): Type
```

Comment:
Same, here are not handle the case or both param are true. (if a real case exists for url)

Reply:
Fixed in 554f6140b

The solution may be "too clever" -- please let me know if you'd prefer a different approach.